### ISSUE TITLE
Veracode Flaw CWE ID 926 Improper Export of Android Application Components

### ISSUE THREAD

**Issue by rajeshwarig** [microsoft/appcenter-sdk-android#1315]
### **Description**

We use the App Center SDK within our Android app. During the Veracode security scans run on the builds, there were 9 security policy violations or flaws reported for App Center. This issue lists 2 flaws that were reported for Veracode CWE ID 926. 

Details for this flaw can be found here: http://cwe.mitre.org/data/definitions/926.html. 

This flaw was identified in line number 1 of DeepLinkActivity.java and line number 1 of DownloadManagerReceiver.java. A screenshot of the flaw listing in Veracode is attached.

![image](https://user-images.githubusercontent.com/1268271/68488461-1220e000-0213-11ea-981e-4651b25d3220.png)

Other Veracode flaws are reported in the following issues:
#1314 
#1316 

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Run Veracode scan to the codebase.

### **Details**

1. Which SDK version are you using?
    - 2.3.0
2. Which OS version did you experience the issue on?
    - N/A
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    -N/A
4. What third party libraries are you using?
    - N/A
5. Please enable verbose logging for your app using `AppCenter.setLogLevel(Log.VERBOSE)` before your call to `AppCenter.start(...)` and include the logs here: N/A

**Comment by guperrot** [comment]
Hi, this one is a false positive.

We do need to redirect back from the browser after open the setup URL, so any activity has to be public we cannot restrict it.

What we did is we validate the Intent data to avoid injecting false redirection data. When we open the browser we pass a unique UUID requestID that our AppCenter portal must use for redirection, if this redirection activity does not use that one time use requestId, we don't process the data so there is no risk here.

As for download completion it's the same, we need to be notified when the download of an APK completes from the download manager and it uses an intent that needs to be public as well. We are checking the intent is for the download we requested for with the download request ID we are expecting. If someone triggers the intent with an invalid download ID we don't process the request. In addition to this check, we are actually checking the download completes and uses the intent only as a ping, so even if knowing the download ID you cannot inject wrong behavior to the SDK.

In short, please mark those 2 issues as false positive.

**Comment by guperrot** [comment]
Hi, closing this as this was answered and not hearing back. Feel free to reopen if you have follow-up questions.
